extensions/cornerstone/src/init.tsx

```diff
@@ -310,37 +316,100 @@ export default async function init({
-
-  //   const cacheKey = `${type}-${id}`;
-  //   if (progress === 0 && !activeToasts.has(cacheKey)) {
-  //     const progressPromise = new Promise((resolve, reject) => {
-  //       activeToasts.set(cacheKey, { resolve, reject });
-  //     });
-
-  //     uiNotificationService.show({
-  //       id: cacheKey,
-  //       title: `${type}`,
-  //       message: `${type}: ${progress}%`,
-  //       autoClose: false,
-  //       promise: progressPromise,
-  //       promiseMessages: {
-  //         loading: `Computing...`,
-  //         success: `Completed successfully`,
-  //         error: 'Web Worker failed',
-  //       },
-  //     });
-  //   } else {
-  //     if (progress === 100) {
-  //       const { resolve } = activeToasts.get(cacheKey);
-  //       resolve({ progress, type });
-  //       activeToasts.delete(cacheKey);
-  //     }
-  //   }
-  // });
+  // Use a single map to track all active worker tasks
+  const activeWorkerTasks = new Map();
+
+  // Create a normalized task key that doesn't include the random ID
+  // This helps us identify and deduplicate the same type of task
+  const getNormalizedTaskKey = type => {
+    return `worker-task-${type.toLowerCase().replace(/\s+/g, '-')}`;
+  };
+
+  eventTarget.addEventListener(EVENTS.WEB_WORKER_PROGRESS, ({ detail }) => {
+    let normalizedKey;
+    let shouldCleanup = false;
+    
+    try {
+      const { progress, type, id } = detail;
+
+      // Skip notifications for compute statistics
+      if (type === cornerstoneTools.Enums.WorkerTypes.COMPUTE_STATISTICS) {
+        return;
+      }
+
+      normalizedKey = getNormalizedTaskKey(type);
+
+      if (progress === 0) {
+        // Check if we're already tracking a task of this type
+        if (!activeWorkerTasks.has(normalizedKey)) {
+          const progressPromise = new Promise((resolve, reject) => {
+            try {
+              activeWorkerTasks.set(normalizedKey, {
+                resolve,
+                reject,
+                originalId: id,
+                type,
+              });
+            } catch (error) {
+              console.error('Error setting active worker task:', error);
+              reject(error);
+            }
+          });
+
+          try {
+            uiNotificationService.show({
+              id: normalizedKey, // Use the normalized key as ID for better deduplication
+              title: `${type}`,
+              message: `Computing...`,
+              autoClose: false,
+              allowDuplicates: false,
+              deduplicationInterval: 60000, // 60 seconds - prevent frequent notifications of same type
+              promise: progressPromise,
+              promiseMessages: {
+                loading: `Computing...`,
+                success: `Completed successfully`,
+                error: 'Web Worker failed',
+              },
+            });
+          } catch (error) {
+            console.error('Error showing web worker notification:', error);
```

Comment:
I could be wrong, but don't we need to (re)throw the error and/or call `reject` here so that the outer `finally` gets called. Apologies if I am missing something here.